test/jira-operations-test.js

```diff
@@ -79,4 +81,29 @@ describe('JIRA Operations Tests', function() {
       assert.equal(result, null);
     });
   });
+
+  describe('Memoization Tests', function() {
+    let spy;
+
+    it('findIssue with Same Key is Called Only Once', () => {
+      spy = sinon.spy(dummyJira, 'findIssue');
+
+      return findParent(dummyJira.issues['WHP-9996'], dummyJira)
+        .then(() => findParent(dummyJira.issues['WHP-9996'], dummyJira))
+        .then(() => {
+          assert.equal(spy.calledOnce, true);
+        });
+    });
+
+    it('listFields with Same JIRA Host is Called Only Once', () => {
+      spy = sinon.spy(dummyJira, 'listFields');
+      dummyJira.host = 'jira.host4.com';
+
+      return getEpicLinkField(dummyJira)
+        .then(() => getEpicLinkField(dummyJira))
+        .then(() =>{
+          assert.equal(spy.calledOnce, true);
+        });
+    });
```

Comment:
We are already testing against the returned value using findParent in jira-operations-test.js